### DISCUSSION TITLE
Highway DQN CNN example - CNN architecture?

### DISCUSSION THREAD

**Discussion by PGomezGonzalo** [eleurent/highway-env#254] [Q&A] [answered]
Hi all.

I am playing a bit with sb3_highway_dqn_cnn.py but I am struggling a lot to understand the architecture of the CNN. 

In line 32 of that script,  a function named DQN is called with an argument:
 model = DQN('CnnPolicy',...)
I have search documentation for that function and I see that function calls for the class [CnnPolicy](https://stable-baselines3.readthedocs.io/en/master/_modules/stable_baselines3/dqn/policies.html#CnnPolicy), which has a parameter called net_arch, but I see it is default to None. 

I also arrived to the [stable_baselines3](https://github.com/DLR-RM/stable-baselines3/blob/master/stable_baselines3/dqn/policies.py#L180) repo, but I can not see there a default architecture. 

I am sorry if this questions does not belong here, but I really could use some help. How the CNN architecture is defined in that example? If it is a default value, where is that default value defined?

I am sorry again if I am missing some important information to answer my question, but I will gladly add whatever you point me it could be useful to answer. 

Thank you in advance!

**Comment by eleurent** [comment, marked as answer]
Hi @PGomezGonzalo 
Here is what I understand from the docs

- `CnnPolicy` has a `net_arch` argument whose default is None, but also a `features_extractor_class` whose default is `NatureCNN`
- `CnnPolicy` inherits from `DQNPolicy`. In its constructor, if `net_arch` is not specified (None), it sets it to a [64, 64] MLP, unless the `features_extractor_class` is set to `NatureCNN` (which is our case), then the `net_arch` is set to [], so we're left with the feature extractor CNN only (no MLP head).
- You can find the NatureCNN implementation [here](https://github.com/DLR-RM/stable-baselines3/blob/master/stable_baselines3/common/torch_layers.py#L51)
